File: src-architecture/game/creator/ApolloSystemCreator.java
package game.creator;

import java.io.IOException;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;

import javax.media.opengl.GL2;

import model.celestials.CelestialBody;
import model.celestials.types.Star;
import model.ship.Ship;
import model.system.StarSystem;
import model.system.parts.SkyBox;
import engine.utils.JsonSystemParser;

/**
 * Creates Apollo system.
 */
public class ApolloSystemCreator extends SystemCreator {

    /* ========== PUBLIC ========== */
    /**
     * Method that creates Apollo system.
     * 
     * TODO In future - systems should be loaded from JSON file.
     */
    public static StarSystem create(GL2 gl) throws IOException {

        List<CelestialBody> bodies = new LinkedList<CelestialBody>();

        Star sun = createStar(gl, bodies);
        createPlanets(gl, bodies, sun);
        SkyBox skyBox = createSkyBox("BlueGalaxy");

        ArrayList<Ship> ships = new ArrayList<Ship>();

        StarSystem system = new StarSystem(skyBox, ships, bodies);
        JsonSystemParser.saveStarSystem("apolloSystem", system);

        return system;
    }

    /* ========== PRIVATE ========== */
    private static Star createStar(GL2 gl, List<CelestialBody> bodies) throws IOException {
        return star(gl, "Apollo", 182671, 0.574849, bodies);
    }

    private static void createPlanets(GL2 gl, List<CelestialBody> bodies, Star sun) throws IOException {
        planet(gl, "Zorh", null, null, 5982.78, 34.92, 0.01, sun, 0.4321, 56.23, 60, bodies);

        planet(gl, "Anthony", null, null, 4329.71, 5.01, 32.41, sun, 0.7218, 43.29, 80, bodies);

        CelestialBody david = planet(gl, "David", null, null, 8123.41, 2120.23, 17.21, sun, 1.3253, 32.12, 100, bodies);
        moon(gl, "Murmillo", null, 3291.89, 8.03, 2.32, david, 0.000329, 1.232, 100, bodies);
        moon(gl, "Retarius", null, 1023.17, 2.75, 4.89, david, 0.000218, 1.452, 80, bodies);

        CelestialBody alicia = planet(gl, "Alicia", null, "AliciaRings", 54495.42, 13.89, 10.03, sun, 4.4171, 18.87,
                120, bodies);
        moon(gl, "Julio", null, 2398.17, 7.81, 11.12, alicia, 0.000318, 1.452, 80, bodies);
        moon(gl, "Maya", null, 1118.81, 3.75, 24.81, alicia, 0.000448, 1.452, 100, bodies);
        moon(gl, "Leo", null, 523.17, 2.75, 8.19, alicia, 0.000398, 1.452, 60, bodies);
        moon(gl, "Julia", null, 872.19, 1.21, 16.91, alicia, 0.000418, 1.452, 80, bodies);

        CelestialBody victor = planet(gl, "Victor", null, "NinurtaRings", 103459.32, 32.43, 19.32, sun, 9.6432,
                12.32, 200, bodies);
        moon(gl, "Hoth", null, 1126.12, 5.81, 12.17, victor, 0.000432, 1.452, 120, bodies);
        moon(gl, "Polie", null, 2926.12, 7.81, 17.17, victor, 0.000518, 1.452, 80, bodies);
        moon(gl, "Barr", null, 676.21, 0.81, 7.11, victor, 0.000628, 1.452, 80, bodies);

        planet(gl, "Ninurta", null, "VictorRings", 62459.32, 10421.43, 65.32, sun, 14.6432, 16.32, 200, bodies);

        wormhole(gl, "solarSystem", 10000.00, 10.43, 0, sun, 4.6432, 10.32, 200, 1, 0, bodies);
    }
}

File: src-architecture/game/creator/ApolloSystemCreator.kt
package game.creator

import model.celestials.CelestialBody
import model.celestials.Star
import model.ship.Ship
import model.system.StarSystem
import java.util.*

class ApolloSystemCreator : SystemCreator() {

    fun create(): StarSystem {
        val bodies = LinkedList<CelestialBody>()

        val sun = createStar(bodies)
        createPlanets(bodies, sun)
        val skyBox = createSkyBox("BlueGalaxy")

        val ships = ArrayList<Ship>()

        return StarSystem(skyBox, ships, bodies)
    }

    private fun createStar(bodies: MutableList<CelestialBody>): Star {
        return star("Apollo", 182671.0F, 1973.20f, bodies)
    }

    private fun createPlanets(bodies: MutableList<CelestialBody>, sun: Star) {
        planet("Zorh", null, null, 5982.78f, 3412.92f, 0.01, sun, 0.4321f, 56.23f, 60F, bodies)

        planet("Anthony", null, null, 4329.71f, 543.01f, 32.41, sun, 0.7218f, 43.29f, 80F, bodies)

        val david = planet("David", null, null, 8123.41f, 2120.23f, 17.21, sun, 1.3253f, 32.12f, 100F, bodies)
        moon("Murmillo", 3291.89F, 829.03f, 2.32, david, 0.00329f, 1.232f, 100F, bodies)
        moon("Retarius", 1023.17F, 281.75f, 4.89, david, 0.00218f, 1.452f, 80F, bodies)

        val alicia = planet("Alicia", null, "AliciaRings", 54495.42f, 13243.89f, 10.03, sun, 4.4171f, 18.87f,
                120F, bodies)
        moon("Julio", 2398.17F, 762.81f, 11.12, alicia, 0.00318f, 1.452f, 80F, bodies)
        moon("Maya", 1118.81F, 342.75f, 24.81, alicia, 0.00448f, 1.452f, 100F, bodies)
        moon("Leo", 523.17F, 21.75f, 8.19, alicia, 0.00398f, 1.452f, 60F, bodies)
        moon("Julia", 872.19F, 162.21f, 16.91, alicia, 0.00418f, 1.452f, 80F, bodies)

        val victor = planet("Victor", null, "NinurtaRings", 103459.32f, 32429.43f, 19.32, sun, 9.6432f,
                12.32f, 200F, bodies)
        moon("Hoth", 1126.12F, 562.81f, 12.17, victor, 0.00432f, 1.452f, 120F, bodies)
        moon("Polie", 2926.12F, 762.81f, 17.17, victor, 0.00518f, 1.452f, 80F, bodies)
        moon("Barr", 676.21F, 72.81f, 7.11, victor, 0.00628f, 1.452f, 80F, bodies)

        planet("Ninurta", null, "VictorRings", 62459.32f, 10421.43f, 65.32, sun, 14.6432f, 16.32f, 200F, bodies)

        wormhole("solarSystem", 10000.00F, 10421.43f, 0.0, sun, 4.6432f, 10.32f, 200F, 1, 0, bodies)
    }
}